src/services/expenses.service.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const { idGenerator } = require('./idGenerator');
+
+let expenses = [];
+
+const getAll = () => {
+  return expenses;
+};
+
+const getById = (id) => {
+  return expenses.find(expense => expense.id === id) || null;
+};
+
+const add = (expense) => {
+  const preparedExpense = {
+    id: idGenerator(expenses),
+    ...expense,
+  };
+
+  expenses.push(preparedExpense);
+
+  return preparedExpense;
+};
+
+const update = (id, expense) => {
+  const expenseToUpdate = expenses.find(user => user.id === id);
+
+  if (!expenseToUpdate) {
+    return;
+  }
+
+  Object.assign(expenseToUpdate, { ...expense });
+
+  return expenseToUpdate;
+};
+
+const remove = (id) => {
+  const isExpenseExists = expenses.some(item => item.id === id);
```

Comment:
[error] should return 404 if user does not exist

Your id and item.id can be different types, try convert it into `number` or `string`. Also try this in other places